pccommon/pccommon/tracing.py

```diff
@@ -249,3 +251,63 @@ def _iter_cql(cql: dict, property_name: str) -> Optional[Union[str, List[str]]]:
                             return result
     # No collection was found
     return None
+
+
+def add_stac_attributes_from_search(search_json: str, request: fastapi.Request) -> None:
+    """
+    Try to add the Collection ID and Item ID from a search to the current span.
+    """
+    collection_id, item_id = parse_collection_from_search(
+        json.loads(search_json), request.method, request.query_params
+    )
+    tracer = Tracer(
+        exporter=exporter,
+        sampler=ProbabilitySampler(1.0),
+    )
+
+    with tracer.span("main") as span:
+        if (
+            hasattr(request.state, "parent_span")
+            and request.state.parent_span is not None
+        ):
+            request.state.parent_span = span
```

Comment:
I'm not super familiar with the opencensus stuff, but this looks like we are creating a new span and overriding the existing one here. I think we should be getting the existing span off of `request.state.parent_span` and using it to set the attribute. Something like

```python
span = request.state.parent_span
span.add_attribute(...)
```

Reply:
Oooo so doing span = request.state.parent_span returns an error saying state does not have a parent_span